features/welcome/components/AbstractWelcomePage.js

```diff
@@ -93,19 +96,69 @@ export class AbstractWelcomePage extends Component {
     _onRoomChange(value) {
         this.setState({ room: value });
     }
+
+    /**
+     * Handler for case when video starts to play.
+     *
+     * @private
+     * @returns {void}
+     */
+    _onVideoPlaying() {
+        let localParticipant = this.props.localParticipant;
+
+        if (localParticipant) {
+            this.props.dispatch(participantVideoStarted(localParticipant.id));
+        }
+    }
+
+    /**
+     * Renders a local video if any.
+     *
+     * @protected
+     * @returns {(ReactElement|null)}
+     */
+    _renderLocalVideo() {
+        let localVideoTrack = this.props.localVideoTrack;
+
+        if (localVideoTrack) {
+            return (
+                <Video
+                    mirror={ shouldMirror(localVideoTrack) }
+                    onPlaying={ this._onVideoPlaying }
```

Comment:
Please perform a case-sensitive whole-word search for onPlaying, then observe that the whole Video along with _onVideoPlaying is duplicated. I know I talked about this when I asked Lance to work on eliminating the duplication that resulted in the introduction of shouldMirror & then I commented that stopping at the introduction of the function is insufficient. Could I please ask you to finish that work by introducing an abstraction that eliminates the new duplication? Could I please ask you to give priority to this fix (along with the other fix I asked you earlier in this PR)?